app/api/ada/hardware-wallet/connectTrezorAdaWallet.js

```diff
@@ -0,0 +1,232 @@
+// @flow
+
+// Handles Connecting Trezor with Yoroi that follow the v2 addressing scheme (bip44)
+
+import _ from 'lodash';
+import { Wallet } from 'rust-cardano-crypto';
+import { getResultOrFail } from '../lib/cardanoCrypto/cryptoUtils';
+import {
+  saveAsAdaAddresses,
+  newAdaAddress
+} from '../adaAddress';
+import {
+  checkAddressesInUse,
+} from '../lib/yoroi-backend-api';
+import type {
+  AdaWallet,
+  AdaHardwareWalletParams
+} from '../adaTypes';
+import {
+  Logger,
+  stringifyError
+} from '../../../utils/logging';
+import { saveCryptoAccount, saveAdaWallet } from '../adaLocalStorage';
+import { createAdaHardwareWallet } from '../adaWallet';
+import { createHardwareWalletAccount } from '../adaAccount';
+import type { ConfigType } from '../../../../config/config-types';
+
+type AddressInfo = { address: string, isUsed: boolean, index: number };
+
+declare var CONFIG: ConfigType;
+const addressScanSize = CONFIG.app.addressScanSize;
+const addressRequestSize = CONFIG.app.addressRequestSize;
+
+/** Creates a new/empty ada wallet
+ * Fetches all related addresses
+ * Caches the fetched address results locally
+ * @returns a new AdaWallet
+ */
+export async function connectTrezorAdaWallet({
+  walletInitData
```

Comment:
I’m confused about what information is passed in walletInitData? We haven’t connected Trezor yet 🤔 so there is no restoration

Reply:
Some of data which is required to make `AdaWallet` object
```
      const walletInitData = {
        cwInitMeta: {
          cwName: walletName,
          cwAssurance: assurance,
          cwUnit: unit
        },
        cwHardwareInfo: {
          vendor: deviceFeatures.vendor,
          model: deviceFeatures.model,
          deviceId: deviceFeatures.device_id,
          label: deviceFeatures.label,
          majorVersion: deviceFeatures.major_version,
          minorVersion: deviceFeatures.minor_version,
          patchVersion: deviceFeatures.patch_version,
          language: deviceFeatures.language,
          publicMasterKey,
        },
      };
```

`connectTrezor` is similar to `restoreWallet`

https://github.com/Emurgo/yoroi-frontend/blob/b6f462b0cce814a2684ef8511ae125754089bb93/app/api/ada/index.js#L377

Reply:
I think we should change the name of `connect` because it’s a little misleading (I got confused). If I just read the name, I would think that the connection is going to be done after calling this method. So it’s strange that the method accepts information that someone would get after calling a *connection* with the hardware wallet.